crates/revm/revm-inspectors/src/access_list.rs

```diff
@@ -0,0 +1,108 @@
+use hashbrown::{HashMap, HashSet};
+use reth_primitives::{AccessList, AccessListItem, Address, H256};
+use revm::{
+    interpreter::{opcode, InstructionResult, Interpreter},
+    Database, EVMData, Inspector,
+};
+
+/// An [Inspector] that collects touched accounts and storage slots.
+///
+/// This can be used to construct an [AccessList] for a transaction via `eth_createAccessList`
+#[derive(Default, Debug)]
+pub struct AccessListInspector {
+    /// All addresses that should be excluded from the final accesslist
+    excluded: HashSet<Address>,
+    /// All addresses and touched slots
+    access_list: HashMap<Address, HashSet<H256>>,
+}
```

Comment:
@joshieDo @onbjerg should we actually just use this for filtering down exactly which slots are accessed on a transaction, and then take the DB pre-transaction and filter down to these slots?